### PR TITLE
support half text & key value props

### PR DESCRIPTION
**Pull request by bermanguy**
image

### CODE REVIEW

File: src/SDK/MonitoringApplianceEmulator/Controls/TextPropertyControl.xaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:local="clr-namespace:Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliancEmulator.Controls"
+             mc:Ignorable="d" 
+             d:DesignHeight="450" d:DesignWidth="800">
+    <Grid>
+        <StackPanel Width="260">
+            <TextBlock Style="{StaticResource OtherDetailsPropertyTitle}"
+                       Text="{Binding Title}"  />
+            <TextBlock Style="{StaticResource AlertDetailsTextPropertyValue}"
+                       Text="{Binding Text}" ToolTip="{Binding Text}" TextTrimming="CharacterEllipsis" />
```
Comment (liadtal) [inline src/SDK/MonitoringApplianceEmulator/Controls/TextPropertyControl.xaml]: >Text [](start = 38, length = 4)

What will happen if the text is long? Will it be wrapped? Can we make this control multiline?

File: src/SDK/MonitoringApplianceEmulator/Controls/KeyValuePropertyControl.xaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:local="clr-namespace:Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliancEmulator"
+             mc:Ignorable="d" 
+             d:DesignHeight="450" d:DesignWidth="800">
+    <UserControl.Resources>
+        <BooleanToVisibilityConverter x:Key="BooleanToVisibilityConverter" />
+    </UserControl.Resources>
+    <DataGrid
+        ItemsSource="{Binding Pairs}"
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Controls/KeyValuePropertyControl.xaml]: >ItemsSource="{Binding Pairs}" [](start = 8, length = 29)

if you're writing this as a control, then you need to expose this as a dependency property - don't assume the view model will have such a property

File: test/SDK/MonitoringApplianceEmulatorTests/ViewModels/AlertDetailsControlViewModelTests.cs [hunk 3]
```diff
@@ -61,87 +61,6 @@ public void WhenCreatingNewViewModelThenItWasInitializedCorrectly()
             Assert.AreEqual("Resource name", alertDetailsControlViewModel.EssentialsSectionProperties[3].ResourceType, "Unexpected essential property 'Resource name'");
             Assert.AreEqual("someVM", alertDetailsControlViewModel.EssentialsSectionProperties[3].ResourceName, "Unexpected essential property 'Resource name'");
 
-            // Verify "Details" properties
-            Assert.AreEqual(6, alertDetailsControlViewModel.DetailsSectionProperties.Count, "Unexpected count of details section properties");
```
Comment (EranGon) [inline test/SDK/MonitoringApplianceEmulatorTests/ViewModels/AlertDetailsControlViewModelTests.cs]: what about adding checks for the PropertiesViewModels?

File: src/SDK/MonitoringApplianceEmulator/Controls/AlertDetailsControl.xaml [hunk 4]
```diff
@@ -42,72 +41,43 @@
...
                 </ListView>
             </Expander>
-            <Expander IsExpanded="True" Style="{StaticResource AlertDetailsExpander}">
+            <Expander IsExpanded="True" Style="{StaticResource AlertDetailsExpander}" >
                 <Expander.Header>
                     <TextBlock Style="{StaticResource AlertDetailsSectionHeader}" Text="OTHER DETAILS"/>
                 </Expander.Header>
-                <ListView ItemsSource="{Binding DetailsSectionProperties}" SelectionMode="Single" BorderThickness="0" 
+                <ListView ItemsSource="{Binding DisplayableProperties}" BorderThickness="0" 
                        Background="{Binding RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Control}}, Path=Background}">
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Controls/AlertDetailsControl.xaml]: >Background [](start = 23, length = 10)

please indent

File: src/SDK/MonitoringApplianceEmulator/Controls/KeyValuePropertyControl.xaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:local="clr-namespace:Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliancEmulator.Controls"
+             mc:Ignorable="d" 
+             d:DesignHeight="450" d:DesignWidth="800">
+    <UserControl.Resources>
+        <BooleanToVisibilityConverter x:Key="BooleanToVisibilityConverter" />
+    </UserControl.Resources>
+    <DataGrid
+        ItemsSource="{Binding Path=KeyValueAlertProperty.Value, Mode=OneWay, RelativeSource={RelativeSource AncestorType=local:KeyValuePropertyControl, AncestorLevel=1}}" 
+        CanUserSortColumns="True"
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Controls/KeyValuePropertyControl.xaml]: >True [](start = 28, length = 4)

False - that's not supported in Ibiza

File: src/SDK/MonitoringApplianceEmulator/Controls/KeyValuePropertyControl.xaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        CanUserAddRows="False"
+        AutoGenerateColumns="False"
+        IsReadOnly="True"
+        Width="260">
+        <DataGrid.Columns>
+            <DataGridTemplateColumn>
+                <DataGridTemplateColumn.Header>
+                    <TextBlock Text="{Binding Path=KeyValueAlertProperty.KeyHeaderName, Mode=OneWay, RelativeSource={RelativeSource AncestorType=local:KeyValuePropertyControl, AncestorLevel=1}}" 
+                        Visibility="{Binding Path=KeyValueAlertProperty.ShowHeaders, Mode=OneWay, RelativeSource={RelativeSource AncestorType=local:KeyValuePropertyControl, AncestorLevel=1}, Converter={StaticResource BooleanToVisibilityConverter}}" />
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Controls/KeyValuePropertyControl.xaml]: >Visibility [](start = 24, length = 10)

indent

File: src/SDK/MonitoringApplianceEmulator/Controls/TextPropertyControl.xaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:local="clr-namespace:Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliancEmulator.Controls"
+             mc:Ignorable="d" 
+             d:DesignHeight="450" d:DesignWidth="800">
+    <Grid>
+        <StackPanel Width="260">
+            <TextBlock Style="{StaticResource OtherDetailsPropertyTitle}"
+                       Text="{Binding Path=TextAlertProperty.DisplayName, Mode=OneWay, RelativeSource={RelativeSource AncestorType=local:TextPropertyControl, AncestorLevel=1}}"  />
+            <TextBlock Style="{StaticResource AlertDetailsTextPropertyValue}"
+                       Text="{Binding Path=TextAlertProperty.Value, Mode=OneWay, RelativeSource={RelativeSource AncestorType=local:TextPropertyControl, AncestorLevel=1}}"
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Controls/TextPropertyControl.xaml]: >Text="{Binding Path=TextAlertProperty.Value, Mode=OneWay, RelativeSource={RelativeSource AncestorType=local:TextPropertyControl, AncestorLevel=1}}"  [](start = 23, length = 148)

what if it's too long? you need to cut it and add "...

**Pull request merged**